src/gui/newwizard/pages/basiccredentialssetupwizardpage.cpp

```diff
@@ -24,30 +24,37 @@ BasicCredentialsSetupWizardPage::BasicCredentialsSetupWizardPage(const QUrl &ser
 {
     _ui->setupUi(this);
 
-    _ui->urlLabel->setText(tr("Connecting to <a href='%1' style='color: %2;'>%1</a>").arg(serverUrl.toString(), Theme::instance()->wizardHeaderTitleColor().name()));
+    const QString linkColor = Theme::instance()->wizardHeaderTitleColor().name();
+
+    _ui->urlLabel->setText(tr("Connecting to <a href='%1' style='color: %2;'>%1</a>").arg(serverUrl.toString(), linkColor));
 
     connect(this, &AbstractSetupWizardPage::pageDisplayed, this, [this]() {
         _ui->usernameLineEdit->setFocus();
     });
 
     // branding
-    switch (Theme::instance()->userIDType()) {
-    case Theme::UserIDUserName:
-        _ui->usernameLabel->setText(tr("Username"));
-        break;
-    case Theme::UserIDEmail:
-        _ui->usernameLabel->setText(tr("E-mail address"));
-        break;
-    case Theme::UserIDCustom:
-        _ui->usernameLabel->setText(Theme::instance()->customUserID());
-        break;
-    default:
-        Q_UNREACHABLE();
-    }
+    const QString usernameLabelText = []() {
```

Comment:
enumToStirng

Reply:
Also, how to handle translations?

Reply:
yes in the theme.
QCoreApplication::translate("enumName","foo")